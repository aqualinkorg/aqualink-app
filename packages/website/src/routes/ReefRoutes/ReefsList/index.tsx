import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  List,
  ListItem,
  ListItemText,
  createStyles,
  withStyles,
  WithStyles,
  Theme,
} from "@material-ui/core";
import { Link } from "react-router-dom";

import {
  reefsListSelector,
  reefsRequest,
} from "../../../store/Reefs/reefsListSlice";

const ReefsList = ({ classes }: ReefsListProps) => {
  const reefsList = useSelector(reefsListSelector);
  const dispatch = useDispatch();

  useEffect(() => {
    if (!reefsList) {
      dispatch(reefsRequest());
    }
  }, [dispatch, reefsList]);

  return (
    <>
      <div className={classes.root}>
        <List component="nav">
          {reefsList?.map((reef) => (
            <Link
              key={`reef-list-item-${reef.id}`}
              style={{ color: "inherit", textDecoration: "none" }}
              to={`/reefs/${reef.id}`}
            >
              <ListItem button>
                <ListItemText style={{ color: "white" }} primary={reef.name} />
              </ListItem>
            </Link>
          ))}
        </List>
      </div>
    </>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    root: {
      width: "100%",
      maxWidth: 360,
      backgroundColor: theme.palette.primary.dark,
    },
  });

type ReefsListProps = WithStyles<typeof styles>;

export default withStyles(styles)(ReefsList);
