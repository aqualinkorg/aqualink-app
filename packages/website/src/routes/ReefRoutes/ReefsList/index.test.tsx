import React from "react";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";

import ReefsList from ".";

const mockStore = configureStore([]);

describe("Reefs List", () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      reefsList: {
        list: [],
        loading: false,
        error: null,
      },
    });

    store.dispatch = jest.fn();

    element = render(
      <Provider store={store}>
        <Router>
          <ReefsList />
        </Router>
      </Provider>
    ).container;
  });

  it("should render with given state from Redux store", () => {
    expect(element).toMatchSnapshot();
  });
});
