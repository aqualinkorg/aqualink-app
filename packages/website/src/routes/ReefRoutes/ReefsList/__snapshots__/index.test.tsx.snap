// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reefs List should render with given state from Redux store 1`] = `
<div>
  <div
    class="ReefsList-root-1"
  >
    <mock-list
      component="nav"
    >
      <a
        href="/reefs/1"
        style="text-decoration: none;"
      >
        <mock-listitem
          button="true"
        >
          <mock-listitemtext
            primary=""
            style="color: white;"
          />
        </mock-listitem>
      </a>
    </mock-list>
  </div>
</div>
`;
